agent/php_observer.c

```diff
@@ -104,4 +113,23 @@ void nr_php_observer_minit() {
   NR_PHP_PROCESS_GLOBALS(orig_execute) = nr_php_observer_no_op;
 }
 
+void nr_throw_exception_hook(zend_object* exception) {
+  zval new_exception;
+  zval* exception_zval = NULL;
+
+  /*
+   * On PHP 7, EG(exception) is stored as a zend_object, and is only
+   * wrapped in a zval when it actually needs to be.
+   */
+  ZVAL_OBJ(&new_exception, exception);
```

Comment:
I'm not clear if the code here is handling PHP 7 differently (based on comment) than other PHP versions since there is no conditional code in this function.  Also all of this code is in a conditional block for PHP >= 8.0 so maybe the comment needs to be adjusted for this code block?

Reply:
Please see pr [e98d885](https://github.com/newrelic/newrelic-php-agent/pull/580/commits/e98d8852f20913be01e41faec16fb1ee794ed576)

Reply:
Thank you!